src/realm/sync/noinst/protocol_codec.cpp

```diff
@@ -9,21 +9,38 @@ using OutputBuffer = util::ResettableExpandableBufferOutputStream;
 
 // Client protocol
 
-void ClientProtocol::make_bind_message(int protocol_version, OutputBuffer& out, session_ident_type session_ident,
-                                       const std::string& server_path, const std::string& signed_user_token,
-                                       bool need_client_file_ident, bool is_subserver)
+void ClientProtocol::make_pbs_bind_message(int protocol_version, OutputBuffer& out, session_ident_type session_ident,
+                                           const std::string& server_path, const std::string& signed_user_token,
+                                           bool need_client_file_ident, bool is_subserver)
 {
     static_cast<void>(protocol_version);
     out << "bind " << session_ident << " " << server_path.size() << " " << signed_user_token.size() << " "
-        << int(need_client_file_ident); // Throws
-    out << " " << int(is_subserver);    // Throws
-    out << "\n";                        // Throws
+        << int(need_client_file_ident) << " " << int(is_subserver) << "\n"; // Throws
     REALM_ASSERT(!out.fail());
 
     out.write(server_path.data(), server_path.size());             // Throws
     out.write(signed_user_token.data(), signed_user_token.size()); // Throws
 }
 
+void ClientProtocol::make_flx_bind_message(int protocol_version, OutputBuffer& out, session_ident_type session_ident,
+                                           const nlohmann::json& json_data, const std::string& signed_user_token,
+                                           bool need_client_file_ident, bool is_subserver)
+{
+    std::string json_data_stg;
+    // Protocol version v8 and above accepts stringified json_data for the first data argument
+    if (protocol_version >= 8 && !json_data.empty()) {
+        json_data_stg = json_data.dump();
+    }
+
+    static_cast<void>(protocol_version);
```

Comment:
no need to static_cast this here. we do it in the pbs function just to avoid an unused variable warning, but we do use it here. we can consider just removing it as a parameter to the pbs version to simplify things a bit.